Cython/Compiler/ExprNodes.py

```diff
@@ -4749,9 +4749,29 @@ def analyse_types(self, env, getting=True):
                 analyse_base=False)
 
         if self.start:
-            self.start = self.start.analyse_types(env)
```

Comment:
This still needs to be done first, because the `None` check is actually only needed if `self.start.is_pyobject` and/or `self.stop.is_pyobject` respectively, which will be known after analysing their types.

Reply:
Ah, very cool - added!